controller/api/locking.py

```diff
@@ -0,0 +1,38 @@
+import logging
+
+from models import get_etcd_client
+
+log = logging.getLogger(__name__)
+LOCK_PATH_TEMPLATE = '/deis/controller/locks/{name}'
+
+
+class AlreadyLocked(Exception):
+    pass
+
+
+def lock(name, ttl=3600):
+    key = LOCK_PATH_TEMPLATE.format(name=name)
+    value = 'locked'
+    client = get_etcd_client()
```

Comment:
I'm not sure that we need to use etcd as our locking mechanism. Why not just use a [Lock](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Lock) instead?